src/engine.rs

```diff
@@ -114,21 +128,14 @@ pub fn run(params: FunctionRunParams) -> Result<FunctionRunResult> {
         export,
         profile_opts,
         scale_factor,
+        engine,
+        module,
     } = params;
 
-    let engine = Engine::new(
-        Config::new()
-            .wasm_multi_memory(true)
-            .wasm_threads(false)
-            .consume_fuel(true)
-            .epoch_interruption(true),
-    )?;
-    let module = Module::from_file(&engine, &function_path)
-        .map_err(|e| anyhow!("Couldn't load the Function {:?}: {}", &function_path, e))?;
-
     let input_stream = MemoryInputPipe::new(input.raw.clone());
     let output_stream = MemoryOutputPipe::new(usize::MAX);
     let error_stream = MemoryOutputPipe::new(usize::MAX);
+    let module = module.as_ref().unwrap();
```

Comment:
If `module` isn't optional, we don't need to do this. I believe we can just use `&module` below.